plugin-api/src/main/java/com/gentics/mesh/plugin/AbstractPluginVerticle.java

```diff
@@ -93,17 +76,22 @@ public File getStorageDir() {
 	@Override
 	public <T> T readConfig(Class<T> clazz) throws FileNotFoundException, IOException {
 		File configFile = getConfigFile();
-		if (!configFile.exists()) {
+		if (!configFile.canRead()) {
 			return null;
 		}
-		try (FileInputStream fis = new FileInputStream(configFile)) {
-			return getYAMLMapper().readValue(fis, clazz);
+		T config = PluginConfigLoader.loadConfig(configFile, null, clazz);
+
+		//try to load local config file
+		File localConfigFile = getLocalConfigFile();
+		if(localConfigFile.canRead()) {
+			config = PluginConfigLoader.loadConfig(localConfigFile, config, clazz);
 		}
+		return config;
 	}
 
 	@Override
 	public <T> T writeConfig(T config) throws IOException {
-		String yaml = getYAMLMapper().writeValueAsString(config);
```

Comment:
Offload whole writeConfig to PluginConfigLoader - The class should also handle write (thus rename the class)